tensordict/prototype/tensorclass.py

```diff
@@ -536,21 +536,38 @@ def __eq__(self, other):
     """Compares the Tensor class object to another object for equality. However, the equality check for non-tensor data is not performed.
 
     Args:
-        other: object to compare to this object
+        other: object to compare to this object. Can be a tensorclass, a
+            tensordict or any compatible type (int, float or tensor), in
+            which case the equality check will be propagated to the leaves.
 
     Returns:
         False if the objects are of different class types, Tensorclass of boolean values for tensor attributes and None for non-tensor attributes
 
     """
-    if not isinstance(other, self.__class__):
+    if not is_tensorclass(other) and not isinstance(
+        other, (TensorDictBase, int, float, Tensor)
```

Comment:
maybe we can use np.isscalar?

Reply:
sure

Reply:
Nope:
```
>>> np.isscalar("x")
True
```

Reply:
from numpy doc of isscalar:
If you need a stricter way to identify a numerical scalar, use isinstance(x, numbers.Number), as that returns False for most non-numerical elements such as strings.

Reply:
i don't see the updated code with the isinstance call. forgot to commit?